pkg/cephmgr/osd/device.go

```diff
@@ -583,16 +583,44 @@ func addOSDAuth(bootstrapConn client.Connection, osdID int, osdDataPath string)
 }
 
 // adds the given OSD to the crush map
-func addOSDToCrushMap(osdConn client.Connection, context *clusterd.Context, osdID int, osdDataPath, location string) error {
-	// get the size of the volume containing the OSD data dir
-	s := syscall.Statfs_t{}
-	if err := syscall.Statfs(osdDataPath, &s); err != nil {
-		return fmt.Errorf("failed to statfs on %s, %+v", osdDataPath, err)
+func addOSDToCrushMap(osdConn client.Connection, context *clusterd.Context, config *osdConfig, location string) error {
+	osdID := config.id
```

Comment:
this is an artifact of changing the function's arguments to be `config *osdConfig` instead of it's individual properties being passed in.   to keep the variable names the same for the rest of the function (and to be sure i didn't miss any!) i just assigned the old argument names to the correct fields on the new `config` argument.